sound/soc/sof/sof-client-probes.c

```diff
@@ -430,7 +416,22 @@ static int sof_probes_client_probe(struct auxiliary_device *auxdev,
 	}
 
 	priv->host_ops = ops;
-	priv->ipc_ops = &ipc3_probe_ops;
+
+	if (sof_client_get_ipc_type(cdev) == SOF_INTEL_IPC4) {
+#if IS_ENABLED(CONFIG_SND_SOC_SOF_INTEL_IPC4)
```

Comment:
<argh, editing comment aftewards and then deciding that it is good as it was end pressing cancel deletes the commtent ... writing the comment again as I remember it>

Yes for ipc type to be SOF_INTEL_IPC4, the CONFIG_SND_SOC_SOF_INTEL_IPC4 has to be enabled, but the refences to IPC4 code has to be enclosed behind #if to avoid undefined symbols, if IPC4 is not enabled. But yes simple #ifdef should do as well. Should I do the switch? Also to sof_client_ipc4_find_module_by_uuid() in sof-client.c?